### PR TITLE
Use core::intrinsics

### PR DESCRIPTION
**Pull request by calebzulawski**
Switches some of the intrinsics for #381. The remainder will be switched after rust-lang/rust#119213

### CODE REVIEW

File: crates/core_simd/tests/pointers.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#![feature(portable_simd, strict_provenance)]
+#![feature(portable_simd, strict_provenance, exposed_provenance)]
```
Comment (programmerjake) [inline crates/core_simd/tests/pointers.rs:1]: Please split this into a separate commit and in the commit message link to the PR where exposed_provenance was split out.

Comment (programmerjake) [reply crates/core_simd/tests/pointers.rs:1]: @bjorn3 this should allow you to drop [the patch](https://github.com/rust-lang/rust/commit/1095e0165c36bd65f5fb3af9955e565137238b65) once this is merged and subtree synced.

Comment (bjorn3) [reply crates/core_simd/tests/pointers.rs:1]: I can actually drop it as soon as this PR is merged. cg_clif's test suite currently directly clones this PR to test it rather than using the copy used by the standard library.

**Review by programmerjake — APPROVED** [review]
